aviator.js

```diff
@@ -1028,6 +1044,23 @@ Route.prototype = {
   **/
   isNamedParam: function (fragment) {
     return fragment.indexOf('/:') === 0;
+  },
+
+  /**
+  Assign the given function to the next sequential key in the this.anonymous
+  object.
+
+  @method generateNotFoundAction
+  @param {Function} func the anonymous function to transform into an action.
+  @return {Object} With a method key that maps to a generated String, and
+  **/
+  anonymousAction: function (func) {
+    var method = 'FUNCTION-' + Object.keys(this.anonymous).length;
```

Comment:
Actually, this could be solved by using a new object for every anonymous function. See [this commit](https://github.com/swipely/aviator/commit/486fdfe3adeedb0ff63ee74307c3b4e8f3874abe).